splash/lua_modules/wraputils.lua

```diff
@@ -134,32 +135,33 @@ end
 -- Handle @lua_property decorators.
 --
 local function setup_property_access(py_object, self, cls)
-  local setters = {}
-  local getters = {}
+  rawset(self, '__getters', {})
+  rawset(self, '__setters', {})
+
   for name, opts in pairs(py_object.lua_properties) do
-    getters[name] = unwraps_python_result(drops_self_argument(py_object[opts.getter]))
+    self.__getters[name] = unwraps_python_result(drops_self_argument(py_object[opts.getter]))
     if opts.setter ~= nil then
-      setters[name] = unwraps_python_result(drops_self_argument(py_object[opts.setter]))
+      self.__setters[name] = unwraps_python_result(drops_self_argument(py_object[opts.setter]))
     else
-      setters[name] = function()
+      self.__setters[name] = function()
         error("Attribute " .. name .. " is read-only.", 2)
       end
     end
   end
 
-  function cls:__newindex(index, value)
-    if setters[index] then
-      return setters[index](self, value)
+  function cls:__index(index)
+    if self.__getters[index] then
+      return self.__getters[index](self)
```

Comment:
I still stumble upon this function's header each time I revisit this code.

Can we rewrite it (and `__newindex` below) to `function cls.__index(self, index)` or `cls.__index = function(self, index)` so that it's more obvious that `self` is an argument to that function rather than `cls` or `self` from the outside scope?